pyogrio/tests/test_geopandas_io.py

```diff
@@ -479,29 +517,126 @@ def test_write_dataframe_gdalparams(tmp_path, naturalearth_lowres):
     assert test_withindex_index_filename.exists() is True
 
 
-def test_write_dataframe_geometry_type(tmp_path, naturalearth_lowres):
+def test_write_dataframe_promote_to_multi_geojson(tmp_path, naturalearth_lowres):
```

Comment:
You might want to then move `promote_to_multi=False` to a dedicated test case for FGB (check for error) and GPKG (check for warning?  can't remember if we catch that).

Reply:
OK.

Regarding promote_to_multi=False, this doesn't give an error anymore but triggers that the layer geometry type will be "Unknown" as was the case in the initial implementation.

Reply:
> For the warning when writing ".gpkg", it isn't caught at the moment, and I don't immediately know if it can be caught as it is - as far as I know - just logging that is written by gdal.

Yes, we don't yet turn it into an actual _python_ warning (I thought we had an issue about it, but don't directly see it), so you can leave this for now (we ran into the same problem for warnings from incorrect file extensions when reading from an in-memory file)